reminder/module.py

```diff
@@ -0,0 +1,465 @@
+import asyncio
+import re
+
+from datetime import datetime, timedelta
+from typing import Optional, List
+
+import nextcord
+from nextcord.ext import commands, tasks
+from nextcord.errors import HTTPException, Forbidden
+
+import dateutil.parser
+
+from pie import check, i18n, logger, utils
+
+from .database import ReminderStatus, ReminderItem
+
+_ = i18n.Translator("modules/reminder").translate
+bot_log = logger.Bot.logger()
+guild_log = logger.Guild.logger()
+
+
+class Reminder(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.reminder.start()
+
+    def cog_unload(self):
+        self.reminder.cancel()
+
+    @tasks.loop(seconds=30)
+    async def reminder(self):
+        max_remind_time = datetime.now() + timedelta(seconds=30)
+
+        items = ReminderItem.get_all(
+            status=ReminderStatus.WAITING, max_remind_date=max_remind_time
+        )
+
+        if items is not None:
+            for item in items:
+                await self._remind(item)
+
+    @reminder.before_loop
+    async def before_reminder(self):
+        print("Reminder loop waiting until ready().")
+        await self.bot.wait_until_ready()
+
+    @commands.guild_only()
+    @commands.cooldown(rate=5, per=20.0, type=commands.BucketType.user)
+    @commands.check(check.acl)
+    @commands.command()
+    async def remindme(
+        self, ctx: commands.Context, datetime_str: str, text: Optional[str]
+    ):
+        text = await self._process_text(ctx, datetime_str, text)
+
+        try:
+            date = utils.time.parse_datetime(datetime_str)
+        except dateutil.parser.ParserError:
+            await ctx.reply(
+                _(
+                    ctx,
+                    "I don't know how to parse `{datetime_str}`, please try again.",
+                ).format(datetime_str=datetime_str)
+            )
+            return
+
+        ReminderItem.add(
+            guild_id=ctx.guild.id,
+            author_id=ctx.author.id,
+            remind_id=ctx.author.id,
+            permalink=ctx.message.jump_url,
+            message=text,
+            origin_date=ctx.message.created_at,
+            remind_date=date,
+        )
+
+        date = date.strftime("%d.%m.%Y %H:%M")
+
+        await bot_log.debug(
+            ctx.author,
+            ctx.channel,
+            f"Reminder created for {ctx.author.name}",
+        )
+
+        await ctx.message.add_reaction("✅")
+        await ctx.message.author.send(
+            _(ctx, "Reminder for you created. Reminder will be sent: {date}").format(
+                date=date
+            )
+        )
+
+    @commands.guild_only()
```

Comment:
I don't know if reminding others should also be a guild only command.

@sinus-x do you have any thoughts?

Reply:
You just don't want to be spammed by anybody with bot reminders. I'd rather let this behind ACL wall.

Reply:
I think it should be limited to guild contexts, you don't want to be spammed by someone you don't even share any guild with.